Fix total_pages for StopSearcher when results are not on a per_page boundary

tmecklem opened the pull request:
Jose Alvarez made this commit awesome.

Signed-off-by: Tim Mecklem <tim@teamgaslight.com>

rockwood commented:
We should add the per_page > 0 check in the valid? method

rockwood commented:
@tmecklem It looks like @geofflane fixed this in a23b53f380b088d0876ece5627e9765c0b620ca0

tmecklem commented:
@rockwood thanks for letting me know about that. I merged @geofflane's changes into the branch. Jose and I were on the fence about failing validation vs. overriding invalid params with the defaults, and I like what Geoff did better than where we last ended up.

I added in the kaminari gem for pagination and changed the StopSearcher to use it per @geofflane's suggestion in Trello.
